augur/refine.py

```diff
@@ -105,17 +105,23 @@ def root_outside_of_treetime(T, root, is_timetree):
     treetime_reroot = None
     assert isinstance(root, list), "Internal augur refine error (requested root should be a list)"
     if root[0] == 'best' and len(root)==1:
+        if remove_outgroup:
+            raise AugurError(f"--remove-outgroup is not valid with {root[0]!r} rooting - you must supply a single strain (outgroup) to root the tree")
```

Comment:
_non-blocking comment_

(Related to your question of [re: new error vs AugurError](https://github.com/nextstrain/augur/pull/1745#discussion_r1948239652) in another PR)

My other use case of custom errors is to use the same error message.
I define the new error with a built-in error message. 
```python
class InvalidUseOfRemoveOutgroup(AugurError):
    """
    Raised when --remove-outgroup is used with an invalid --root option.
    """
    def __init__(self, root):
        self.root = root

    def __str__(self):
        return f"--remove-outgroup is not valid with {self.root!r} rooting - you must supply a single strain (outgroup) to root the tree"
```

Then just raise the error
```python
if remove_outgroup:
    raise InvalidUseOfRemoveOutgroup(root[0])
```

Reply:
I hadn't thought of this use - nice. I've refactored the PR to use it. Thanks!